Explorer/Assets/DCL/Backpack/Gifting/Presenters/GiftSelection/GiftingHeaderPresenter.cs

```diff
@@ -0,0 +1,126 @@
+        private readonly UserWalletAddressElementController walletAddressController;
+        private readonly IInputBlock inputBlock;
+
+        private CancellationTokenSource? searchCts;
+
+        public Sprite? CurrentRecipientAvatarSprite
+        {
+            get;
+            set;
+        }
+
+        public GiftingHeaderPresenter(GiftingHeaderView view,
+            IProfileRepository profileRepository,
+            ProfileRepositoryWrapper  profileRepositoryWrapper,
+            IInputBlock inputBlock)
+        {
+            this.view = view;
+            this.profileRepository = profileRepository;
+            this.profileRepositoryWrapper = profileRepositoryWrapper;
+            this.inputBlock = inputBlock;
+
+            walletAddressController = new UserWalletAddressElementController(view.UserProfileWallet);
+
+            view.SearchBar.inputField.onSelect.AddListener(OnSearchSelected);
+            view.SearchBar.inputField.onDeselect.AddListener(OnSearchDeselected);
+            view.SearchBar.inputField.onValueChanged.AddListener(DebounceSearch);
+            view.SearchBar.clearSearchButton.onClick.AddListener(ClearSearch);
+        }
+
+        private void OnSearchSelected(string text)
+        {
+            inputBlock.Disable(InputMapComponent.Kind.SHORTCUTS,
+                InputMapComponent.Kind.IN_WORLD_CAMERA,
+                InputMapComponent.Kind.PLAYER);
+        }
+
+        private void OnSearchDeselected(string text)
+        {
+            inputBlock.Enable(InputMapComponent.Kind.SHORTCUTS,
+                InputMapComponent.Kind.IN_WORLD_CAMERA,
+                InputMapComponent.Kind.PLAYER);
+        }
+
+        public async UniTask SetupAsync(string userId, string username, CancellationToken ct)
+        {
+            // Fetch the full profile of the person we are gifting to
+            var profile = await profileRepository.GetAsync(userId, 0, ct: ct);
+            if (profile == null) return;
+
+            // Set the main title text
+            var userNameColor = profile.UserNameColor;
+            string? userNameColorHex = ColorUtility.ToHtmlStringRGB(userNameColor);
+            view.Title.text = $"Send a Gift to <color=#{userNameColorHex}>{profile.Name}</color>";
+
+            // Use the existing ProfilePictureView's setup logic
+            // Note: The 'Setup' method is obsolete, but we follow the existing pattern for now.
+            // You will need a reference to a ProfileRepositoryWrapper if that's what your project uses.
+            // Let's assume you can get one or adapt this call to the modern `Bind` method later.
+            view.UserProfileImage.Setup(profileRepositoryWrapper,
+                profile.UserNameColor,
+                profile.Avatar.FaceSnapshotUrl);
+
+            // Aleary cached
+            CurrentRecipientAvatarSprite =  profileRepositoryWrapper.GetProfileThumbnail(profile.Avatar.FaceSnapshotUrl);
```

Comment:
I don't see a guarantee here to be cached as you don't `await` the previous method. If you convert your logic to binding you can subscribe to the thumbnail like this: `viewModelProp.Subscribe(OnThumbnailUpdated);`. And you need to initiate such subscription only once on the class level.